File: media-docs/app-icons/reference/AppIconOption.swift
import Foundation

enum AppIconOption: String, CaseIterable {
    case `default`
    case dark
    case clearLight
    case clearDark
    case tintedLight
    case tintedDark

    var alternateIconName: String? {
        switch self {
        case .default:
            return nil
        case .dark:
            return "AppIconDark"
        case .clearLight:
            return "AppIconClearLight"
        case .clearDark:
            return "AppIconClearDark"
        case .tintedLight:
            return "AppIconTintedLight"
        case .tintedDark:
            return "AppIconTintedDark"
        }
    }

    static func fromAlternateIconName(_ name: String?) -> AppIconOption {
        switch name {
        case "AppIconDark":
            return .dark
        case "AppIconClearLight":
            return .clearLight
        case "AppIconClearDark":
            return .clearDark
        case "AppIconTintedLight":
            return .tintedLight
        case "AppIconTintedDark":
            return .tintedDark
        default:
            return .default
        }
    }
}

File: HomelabAndroid/app/src/main/java/com/homelab/app/util/AppIconOption.kt
package com.homelab.app.util

import android.content.ComponentName
import androidx.annotation.DrawableRes
import androidx.annotation.StringRes
import com.homelab.app.R

enum class AppIconOption(
    val persistedValue: String,
    val aliasName: String,
    @DrawableRes val previewDrawableRes: Int,
    @StringRes val labelRes: Int
) {
    DEFAULT(
        persistedValue = "default",
        aliasName = ".launcher.DefaultIconAlias",
        previewDrawableRes = R.drawable.ic_launcher_foreground_default,
        labelRes = R.string.settings_app_icon_default
    ),
    DARK(
        persistedValue = "dark",
        aliasName = ".launcher.DarkIconAlias",
        previewDrawableRes = R.drawable.ic_launcher_foreground_dark,
        labelRes = R.string.settings_app_icon_dark
    ),
    CLEAR_LIGHT(
        persistedValue = "clear_light",
        aliasName = ".launcher.ClearLightIconAlias",
        previewDrawableRes = R.drawable.ic_launcher_foreground_clear_light,
        labelRes = R.string.settings_app_icon_clear_light
    ),
    CLEAR_DARK(
        persistedValue = "clear_dark",
        aliasName = ".launcher.ClearDarkIconAlias",
        previewDrawableRes = R.drawable.ic_launcher_foreground_clear_dark,
        labelRes = R.string.settings_app_icon_clear_dark
    ),
    TINTED_LIGHT(
        persistedValue = "tinted_light",
        aliasName = ".launcher.TintedLightIconAlias",
        previewDrawableRes = R.drawable.ic_launcher_foreground_tinted_light,
        labelRes = R.string.settings_app_icon_tinted_light
    ),
    TINTED_DARK(
        persistedValue = "tinted_dark",
        aliasName = ".launcher.TintedDarkIconAlias",
        previewDrawableRes = R.drawable.ic_launcher_foreground_tinted_dark,
        labelRes = R.string.settings_app_icon_tinted_dark
    );

    fun componentName(packageName: String): ComponentName {
        return ComponentName(packageName, packageName + aliasName)
    }

    companion object {
        fun fromPersistedValue(value: String?): AppIconOption {
            if (value.isNullOrBlank()) return DEFAULT
            return entries.firstOrNull { it.persistedValue == value } ?: DEFAULT
        }
    }
}
